rustls-mbedcrypto-provider/src/self_tests.rs

```diff
@@ -0,0 +1,173 @@
+            0x87, 0x18, 0x58, 0x65
+        ]
+    );
+}
+
+// test copied from rustls repo
+/// TLS 1.3 KDF [test case 2](https://datatracker.ietf.org/doc/html/rfc5869#appendix-A.2).
+pub fn tls13_kdf_test_case_2() {
+    let hkdf = HkdfUsingHmac(&super::hmac::HMAC_SHA256);
+    let ikm: Vec<u8> = (0x00u8..=0x4f).collect();
+    let salt: Vec<u8> = (0x60u8..=0xaf).collect();
+    let info: Vec<u8> = (0xb0u8..=0xff).collect();
+
+    let output: [u8; 82] = expand(
+        hkdf.extract_from_secret(Some(&salt), &ikm)
+            .as_ref(),
+        &[&info],
+    );
+
+    assert_eq!(
+        &output,
+        &[
+            0xb1, 0x1e, 0x39, 0x8d, 0xc8, 0x03, 0x27, 0xa1, 0xc8, 0xe7, 0xf7, 0x8c, 0x59, 0x6a, 0x49, 0x34, 0x4f, 0x01, 0x2e,
+            0xda, 0x2d, 0x4e, 0xfa, 0xd8, 0xa0, 0x50, 0xcc, 0x4c, 0x19, 0xaf, 0xa9, 0x7c, 0x59, 0x04, 0x5a, 0x99, 0xca, 0xc7,
+            0x82, 0x72, 0x71, 0xcb, 0x41, 0xc6, 0x5e, 0x59, 0x0e, 0x09, 0xda, 0x32, 0x75, 0x60, 0x0c, 0x2f, 0x09, 0xb8, 0x36,
+            0x77, 0x93, 0xa9, 0xac, 0xa3, 0xdb, 0x71, 0xcc, 0x30, 0xc5, 0x81, 0x79, 0xec, 0x3e, 0x87, 0xc1, 0x4c, 0x01, 0xd5,
+            0xc1, 0xf3, 0x43, 0x4f, 0x1d, 0x87
+        ]
+    );
+}
+
+// test copied from rustls repo
+/// TLS 1.3 KDF [test case 3](https://datatracker.ietf.org/doc/html/rfc5869#appendix-A.3).
+pub fn tls13_kdf_test_case_3() {
+    let hkdf = HkdfUsingHmac(&super::hmac::HMAC_SHA256);
+    let ikm = &[0x0b; 22];
+    let salt = &[];
+    let info = &[];
+
+    let output: [u8; 42] = expand(
+        hkdf.extract_from_secret(Some(salt), ikm)
+            .as_ref(),
+        info,
+    );
+
+    assert_eq!(
+        &output,
+        &[
+            0x8d, 0xa4, 0xe7, 0x75, 0xa5, 0x63, 0xc1, 0x8f, 0x71, 0x5f, 0x80, 0x2a, 0x06, 0x3c, 0x5a, 0x31, 0xb8, 0xa1, 0x1f,
+            0x5c, 0x5e, 0xe1, 0x87, 0x9e, 0xc3, 0x45, 0x4e, 0x5f, 0x3c, 0x73, 0x8d, 0x2d, 0x9d, 0x20, 0x13, 0x95, 0xfa, 0xa4,
+            0xb6, 0x1a, 0x96, 0xc8
+        ]
+    );
+}
+
+/// Run all the self_tests. If any test fails, this function will panic.
+///
+/// If `verbose` is true, print messages about tests that have been executed.
+pub fn self_tests(verbose: bool) {
+    macro_rules! print_msg {
+        ($($tt: tt)*) => {
+            if verbose { std::println!($($tt)*) }
+        };
+    }
+
+    #[cfg(feature = "tls12")]
+    {
+        tls12_sha256_prf_test_1();
+        print_msg!("tls12_sha256_prf_test_1 passed.");
+        tls12_sha256_prf_test_2();
+        print_msg!("tls12_sha256_prf_test_2 passed.");
+        tls12_sha384_prf_test();
+        print_msg!("tls12_sha384_prf_test passed.");
+    }
+
+    tls13_kdf_test_case_1();
+    print_msg!("tls13_kdf_test_case_1 passed.");
+    tls13_kdf_test_case_2();
+    print_msg!("tls13_kdf_test_case_2 passed.");
+    tls13_kdf_test_case_3();
+    print_msg!("tls13_kdf_test_case_3 passed.");
+}
```

Comment:
Should we have one line for printing all tests are passed, which will be helpful for FIPS validation..

Reply:
Good idea!